tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaTwoParamsAsync.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="LambdaTwoParamsAsync.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS
+{
+    /// <summary>
+    /// Lambda customer handler calltarget instrumentation
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class LambdaTwoParamsAsync
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TArg1">Type of the incommingEvent</typeparam>
+        /// <typeparam name="TArg2">Type of the context</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="incommingEvent">IncommingEvent value</param>
+        /// <param name="context">Context value.</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TArg1, TArg2>(TTarget instance, TArg1 incommingEvent, TArg2 context)
+        {
+            Serverless.Debug("OnMethodBeginOK - two params");
+            return LambdaCommon.StartInvocation(incommingEvent, new LambdaRequestBuilder());
```

Comment:
Can you/is it worth caching this object? Not sure if that makes sense or not in a serverless context 🙂

Reply:
✅